test/clj/rems/api/test_api.clj

```diff
@@ -25,3 +25,26 @@
                        (authenticate "42" "handler")
                        handler)]
           (is (response-is-not-found? resp)))))))
+
+(deftest test-api-key-roles
+  (testing "API key roles"
+    (testing "all available"
+      (let [resp (-> (request :get "/api/forms")
+                     (authenticate "42" "owner")
+                     handler)]
+        (is (= 200 (:status resp)))))
+    (testing "handler and owner roles unavailable"
+      (let [resp (-> (request :get "/api/forms")
+                     (authenticate "43" "owner")
+                     handler)]
+        (is (response-is-forbidden? resp)))))
+  (testing ":api-key role not available when using wrong API key"
```

Comment:
There was a bug previously, allowing the user to obtain :api-key role when using wrong API key but authenticating with cookie and csrf-token. Fixed in this patch, and added this test.

Reply:
good catch